docker/check-cpu.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+
+# This script is used to check the amount of available memory to ensure 2 GB of memory per process.
+# If the number of processes is greater than the amount of available memory (considering 2 GB per core) we restrict the number of cores to use.
+#
+# Sample usage:
+# CPU_CORES=`check-cpu.sh`
+# make -j${CPU_CORES}
+
+
+NPROC=$(nproc)
+
+FREE_MEM=$(free -g | grep Mem | awk '
```

Comment:
`FREE_RAM=$(awk '/^MemAvailable:/{printf("%d",$2/1024/1024)}' /proc/meminfo)
`

Reply:
@sanchayanghosh What do you think?